### DISCUSSION TITLE
Jellyfin automate certificate

### DISCUSSION THREAD

**Discussion by PutoPunko** [alexbelgium/hassio-addons#1627] [Q&A] [answered]
Hi,

I am using HA OS (core 2024.10.4 supervisor 2024.10.3 OS 13.2). I followed isntructions and generated the pfx certificate for https connection. However I have to manually execute openssl every time certificate expires. I tried to create a shell command to execute openssl automatically to renew certificate (using Duck DNS Let's encrypt pem files) however it is not working (I suppose it is a matter of permissions as the command works on SSH addon).

Has someone found a solution to generate pfx certificate (from pem) automatically?

Thanks,

**Comment by alexbelgium** [comment, marked as answer]
Hi, you can use this to auto execute code at add-on start https://github.com/alexbelgium/hassio-addons/wiki/Add%E2%80%90ons-feature-:-customisation

**Reply by PutoPunko** [reply]
Hi,

It worked after I output pfx creation to jellyfin folder.

Thanks!!
